### PR TITLE
Updating type definitions to add IDBBatchAtomicVFS example

### LINKED ISSUE #218: Missing updated types for examples
When attempting to use some of the examples from a Typescript app, the type information is outdated or missing. For example:

```typescript
// this is missing type info and typescript complains
import { IDBBatchAtomicVFS } from "wa-sqlite/src/examples/IDBBatchAtomicVFS.js";
```

I can get this work locally by adding to the `index.d.ts`:

```typescript
** @ignore */
declare module 'wa-sqlite/src/examples/IDBBatchAtomicVFS.js' {
  import * as VFS from "wa-sqlite/src/VFS.js";

  export interface VFSOptions {
    durability?: "default" | "strict" | "relaxed";
    purge?: "deferred" | "manual";
    purgeAtLeast?: number;
  }

  export class IDBBatchAtomicVFS extends VFS.Base {
    constructor(idbDatabaseName:string, options: VFSOptions)
  }
}
```

Happy to make a PR if you'd like, but I was unsure if that `index.d.ts` was generated or not.

### PR DESCRIPTION
**Pull request by thegoldenmule**
### Checklist
- [x] I grant to recipients of this Project distribution a perpetual,
non-exclusive, royalty-free, irrevocable copyright license to reproduce, prepare
derivative works of, publicly display, sublicense, and distribute this
Contribution and such derivative works.
- [x] I certify that I am legally entitled to grant this license, and that this
Contribution contains no content requiring a license from any third party.

Fix #218.

### CODE REVIEW

File: src/types/index.d.ts [hunk 1]
```diff
@@ -1315,3 +1315,18 @@ declare module 'wa-sqlite/src/examples/tag.js' {
    */
   export function tag(sqlite3: any, db: number): (arg0: TemplateStringsArray, ...args: any[]) => Promise<object[]>;
 }
+
+/** @ignore */
+declare module 'wa-sqlite/src/examples/IDBBatchAtomicVFS.js' {
+  import * as VFS from "wa-sqlite/src/VFS.js";
+
+  export interface VFSOptions {
```
Comment (rhashimoto) [inline src/types/index.d.ts:1323]: What is the scope of the `VFSOptions` interface name? If it's global, i.e. declaring `VFSOptions` in a different module would conflict, then maybe `IDBBatchAtomicVFSOptions` would be better.

Comment (thegoldenmule) [reply src/types/index.d.ts]: @rhashimoto Sure, I've renamed it.

**Comment by thegoldenmule** [comment]
Actually, I'm sorta confused.

@rhashimoto looks like the latest version of this library on npm is 1.0.0 (https://www.npmjs.com/package/wa-sqlite).

There is also a package called `wa-sqlite-db` that is 1.0.2.

However, the `IDBBatchAtomicVFS` in both of those has a completely different interface than the `IDBBatchAtomicVFS` in the `main` branch here (1.0.3). Is it possible you could republish the latest package?

**Comment by rhashimoto** [comment]
I don't publish wa-sqlite to npm. There is no official package there.

**Comment by thegoldenmule** [comment]
Ah, well that explains it.

**Comment by thegoldenmule** [comment]
This isn't relevant. I didn't realize this wasn't on NPM, so I've been working with an older version.

**Pull request closed without merging**